lib/project_types/theme/themekit.rb

```diff
@@ -3,53 +3,67 @@ class Themekit
     THEMEKIT = File.join(ShopifyCli.cache_dir, "themekit")
 
     class << self
-      def create(ctx, password:, store:, name:)
-        stat = ctx.system(THEMEKIT,
-                          'new',
-                          "--password=#{password}",
-                          "--store=#{store}",
-                          "--name=#{name}")
+      def create(ctx, password:, store:, name:, env:)
+        command = build_command('new', env)
+        command << "--password=#{password}"
+        command << "--store=#{store}"
+        command << "--name=#{name}"
+
+        stat = ctx.system(command.join(' '))
         stat.success?
       end
 
-      def deploy(ctx)
-        unless push(ctx)
+      def deploy(ctx, env:)
+        flags = '--env=' + env if env
+        unless push(ctx, flags: flags)
           ctx.abort(ctx.message('theme.deploy.push_fail'))
         end
         ctx.done(ctx.message('theme.deploy.info.pushed'))
 
-        stat = ctx.system(THEMEKIT, 'publish')
+        command = build_command('publish', env)
+        stat = ctx.system(command.join(' '))
         stat.success?
       end
 
       def ensure_themekit_installed(ctx)
         Tasks::EnsureThemekitInstalled.call(ctx)
       end
 
-      def pull(ctx, store:, password:, themeid:)
-        stat = ctx.system(THEMEKIT,
-                          "get",
-                          "--store=#{store}",
-                          "--password=#{password}",
-                          "--themeid=#{themeid}")
+      def pull(ctx, store:, password:, themeid:, env:)
+        command = build_command('get', env)
+        command << "--password=#{password}"
+        command << "--store=#{store}"
+        command << "--themeid=#{themeid}"
+
+        stat = ctx.system(command.join(' '))
         stat.success?
       end
 
       def push(ctx, files: nil, flags: nil, remove: false)
-        command = [THEMEKIT]
-        command << (remove ? 'remove' : 'deploy')
+        command = [THEMEKIT, (remove ? 'remove' : 'deploy')]
```

Comment:
Any particular reason for this command not to call `build_command`?

Reply:
In `build_command`, we specify the env, which this command does as a flag with the other flags. Would it be better to use `build_command`, leave the env as `nil` every time, and then pass the env as a flag if needed? Or perhaps separate `--env` from the rest of the flags in the command?

Reply:
One option I can think of would be to, inside `Themekit.push`, extract the env flag from `flags`, pass it into `build_command` and then add the other flags.

It's jumping through a few more hoops, but it keeps our handling of the env flag consistent throughout all commands, and it ensures that we run any other logic we may need to in the future for `push` as well. What do you think?

Reply:
For greater consistency across commands, it would be best to separate the env flag before calling `Themekit.push` and pass the env as a separate parameter; this makes the command messier, but follows the pattern. How should these two factors be prioritized?

Reply:
That makes sense. Perhaps making the 'core' methods more uniform is the best way to go here. At any rate, this isn't a huge deal and we're basically extracting an item from an array, not such a horrible thing to do 😄